cors.go

```diff
@@ -100,13 +103,46 @@ func (c Config) Validate() error {
 		return errors.New("conflict settings: all origins disabled")
 	}
 	for _, origin := range c.AllowOrigins {
-		if origin != "*" && !c.validateAllowedSchemas(origin) {
-			return errors.New("bad origin: origins must either be '*' or include " + strings.Join(c.getAllowedSchemas(), ","))
+		if !strings.Contains(origin, "*") && !c.validateAllowedSchemas(origin) {
+			return errors.New("bad origin: origins must contain '*' or include " + strings.Join(c.getAllowedSchemas(), ","))
 		}
 	}
 	return nil
 }
 
+func (c Config) parseWildcardRules() [][]string {
+	var wRules [][]string
+
+	if !c.AllowWildcard {
+		return wRules
+	}
+
+	for _, o := range c.AllowOrigins {
+		if !strings.Contains(o, "*") {
+			continue
+		}
+
+		if c := strings.Count(o, "*"); c > 1 {
+			panic(errors.New("only one * is allowed").Error())
+		}
+
+		i := strings.Index(o, "*")
+		if i == 0 {
+			wRules = append(wRules, []string{"*", o[1:]})
+			continue
+		}
+		if i == len(o) {
+			wRules = append(wRules, []string{o[:i-1], "*"})
+			continue
+		}
+		if i != 0 && i != len(o) {
```

Comment:
the `if` statement is invalid

Reply:
why? it's situation when ```*``` symbol is in the middle of the string, https://api.*.somedomain.com for example.

Reply:
btw I can get rid of this ```if``` because there is no reason for double check, previous cases when ```*``` is in the beginning and in the end of the string are already handled.